model/src/main/kotlin/VcsInfo.kt

```diff
@@ -124,8 +125,20 @@ data class VcsInfo(
 }
 
 class VcsInfoDeserializer : StdDeserializer<VcsInfo>(VcsInfo::class.java) {
+    companion object {
+        val KNOWN_FIELDS = listOf("type", "url", "revision", "resolvedRevision", "path", "data")
```

Comment:
How about using reflection instead? We could iterate over all fields in the node, and fail if one of their name is not in the property names of [memberProperties](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.reflect.full/member-properties.html) for `VcsInfo`.

Reply:
Done.